File: app/src/main/java/com/harry/pullgo/CalendarEventDecorator.java
package com.harry.pullgo;

import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;
import com.prolificinteractive.materialcalendarview.spans.DotSpan;

import java.util.Collection;
import java.util.HashSet;

public class CalendarEventDecorator implements DayViewDecorator {
    private final int color;
    private final HashSet<CalendarDay> dates;

    public CalendarEventDecorator(int color, Collection<CalendarDay> dates){
        this.color=color;
        this.dates=new HashSet<>(dates);
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return dates.contains(day);
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.addSpan(new DotSpan(5,color));
    }
}

File: app/src/main/java/com/ich/pullgo/presentation/main/common/components/calendar_screen/util/CalendarEventDecorator.kt
package com.ich.pullgo.presentation.main.common.components.calendar_screen.util

import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade
import com.prolificinteractive.materialcalendarview.spans.DotSpan
import java.util.*

class CalendarEventDecorator(private val color: Int, dates: Collection<CalendarDay>?) : DayViewDecorator {
    private val dates: HashSet<CalendarDay> = HashSet(dates)
    
    override fun shouldDecorate(day: CalendarDay): Boolean {
        return dates.contains(day)
    }

    override fun decorate(view: DayViewFacade) {
        view.addSpan(DotSpan(8F, color))
    }

}